test/test_torch.py

```diff
@@ -10033,12 +10039,15 @@ def test_ops(op, string_of_function_name, expected_output):
             # Check that output maintained correct shape
             self.assertEqual(raw_tensor.shape, raw_tensor.grad.shape)
 
+        expected_str_out = '''tensor([4., inf, inf, inf, inf, nan, nan])'''
```

Comment:
`self.assertEqual` with `allow_inf=True` should check locations of `nan` and `inf`, so should be good enough to compare. Comparing results with a string is very fragile.